src/pages/howManyProducts.tsx

```diff
@@ -0,0 +1,93 @@
+    numberOfProductsInput?: string;
+};
+
+type HowManyProductProps = {
+    pageHeadingMessage: string;
+    inputCheck: InputCheck;
+    errors: ErrorInfo[];
+};
+
+const HowManyProducts = ({ pageHeadingMessage, inputCheck, errors }: HowManyProductProps): ReactElement => (
+    <Layout title={buildTitle(errors, title)} description={description}>
+        <main className="govuk-main-wrapper app-main-class" id="main-content" role="main">
+            <form action="/api/howManyProducts" method="post">
+                <ErrorSummary errors={errors} />
+                <div className={`govuk-form-group${inputCheck?.error ? ' govuk-form-group--error input-error' : ''}`}>
+                    <fieldset className="govuk-fieldset" aria-describedby="page-heading">
+                        <legend className="govuk-fieldset__legend govuk-fieldset__legend--xl">
+                            <h1 className="govuk-fieldset__heading" id="page-heading">
+                                {pageHeadingMessage}
+                            </h1>
+                        </legend>
+                        <label className="govuk-label" htmlFor="numberOfProducts">
+                            Number of fare products (up to a maximum of 10)
+                        </label>
+                        {inputCheck?.error ? (
+                            <span id="numberOfProducts-error" className="govuk-error-message">
+                                <span className="govuk-visually-hidden">Error:</span> {inputCheck.error}
+                            </span>
+                        ) : null}
+                        <input
+                            className={`govuk-input govuk-input--width-2 ${
+                                inputCheck?.error ? 'govuk-input--error' : ''
+                            }`}
+                            id="numberOfProducts"
+                            name="numberOfProductsInput"
+                            type="text"
+                            defaultValue={!inputCheck?.error ? inputCheck?.numberOfProductsInput : ''}
+                            aria-describedby={inputCheck?.error ? `numberOfProducts-error` : ''}
+                        />
+                    </fieldset>
+                </div>
+                <input
+                    type="submit"
+                    value="Continue"
+                    id="continue-button"
+                    className="govuk-button govuk-button--start"
+                />
+            </form>
+        </main>
+    </Layout>
+);
+
+export const getDynamicPageHeadingMessage = (cookies: { [key: string]: string }): string => {
+    let pageHeadingMessage = 'How many products do you have for this zone or selected services?';
+    if (cookies[CSV_ZONE_UPLOAD_COOKIE]) {
+        pageHeadingMessage = 'How many products do you have for this zone?';
+    }
+    if (!cookies[CSV_ZONE_UPLOAD_COOKIE]) {
+        pageHeadingMessage = 'How many products do you have for your selected services?';
```

Comment:
Could use template literals and have 

pageHeadingMessage = `How many products do you have for ${selected services? cookies[CSV_ZONE_UPLOAD_COOKIE]} ? this zone :  your selected services?'`;

it seems either way the message will change

Reply:
Good catch! I did have that originally but had complaints from eslint. Thinking back though this might have been because I was trying to chain too many ternary operators